src/state.cairo

```diff
@@ -46,22 +50,34 @@ pub struct Header {
     pub nonce: u32,
 }
 
-/// Transaction
-/// https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format
+/// Extended transaction.
+/// https://learnmeabitcoin.com/technical/transaction/
+///
+/// Contains additional "meta" fields required for validation.
 #[derive(Drop, Copy)]
 pub struct Transaction {
     /// The version of the transaction.
     pub version: i32,
+    /// Flag which indicates the presence of witness data.
+    /// Segwit marker and flag do not contribute to TXID (transaction hash),
+    /// but do contribute to wTXID.
+    pub is_segwit: bool,
     /// The inputs of the transaction.
     pub inputs: Span<TxIn>,
     /// The outputs of the transaction.
     pub outputs: Span<TxOut>,
+    /// The list of witnesses, one for each input.
+    /// Each witness is a list of elements that are to be pushed onto stack.
+    /// Witnesses do not contribute to TXID but do contribute to wTXID.
+    pub witnesses: Span<Span<ByteArray>>,
     /// The lock time of the transaction.
     pub lock_time: u32,
+    /// Transaction fee which is diff between total input and output amounts (meta field)
+    pub fee: i64,
```

Comment:
Shouldn't it be just calculated from inputs and outputs?

Reply:
It's used in two validations, first when we check that transaction spends no more than it can (transaction validation), second time when we check the coinbase amount (block validation).
Having this field cached is handy, the alternative is returning it from transaction validation function (like it's done in Zerosync) but it's not very clean.

Reply:
I would keep it as a temp variable returned by a validation function. Adding it to a structure that is supposed to be an read only input seems confusing to me.

Reply:
Ok!